learning_observer/learning_observer/incoming_student_event.py

```diff
@@ -345,12 +347,19 @@ async def incoming_websocket_handler(request):
     event_metadata['source'] = first_event['source']
 
     # We authenticate the student
-    event_metadata['auth'] = await learning_observer.auth.events.authenticate(
-        request=request,
-        headers=header_events,
-        first_event=first_event,  # This is obsolete
-        source=json_msg['source']
-    )
+    try:
+        event_metadata['auth'] = await learning_observer.auth.events.authenticate(
+            request=request,
+            headers=header_events,
+            first_event=first_event,  # This is obsolete
+            source=json_msg['source']
+        )
+    except aiohttp.web.HTTPForbidden as e:
+        auth_response = json.loads(e.reason)
+        # Send the status message to the client (chrome extension)
+        await ws.send_json({"status": auth_response.get('type')})
+        debug_log(auth_response.get('msg'))
+        return ws
```

Comment:
should we close the connection after they failed to authenticate? Does this keep the connection open? Detailing the workflow in some fashion would be useful for future devs to understand.

Reply:
Yes, we keep the connection open and I have put a reason for that in a comment (temporary forbidden access will need to reauthenticate in the future and any other need for reauthentication too)